src/plugins/data/server/search/session/types.ts

```diff
@@ -17,19 +17,32 @@
  * under the License.
  */
 
-import { Observable } from 'rxjs';
-import { CoreStart, KibanaRequest } from 'kibana/server';
-import { ISearchStrategy } from '../types';
-import { IKibanaSearchRequest, IKibanaSearchResponse } from '../../../common/search';
+import {
+  CoreStart,
+  KibanaRequest,
+  SavedObject,
+  SavedObjectsFindOptions,
+  SavedObjectsFindResponse,
+  SavedObjectsUpdateResponse,
+} from 'kibana/server';
+import { IKibanaSearchRequest, ISearchOptions } from '../../../common/search';
 
-export interface IScopedSessionService {
-  search: <Request extends IKibanaSearchRequest, Response extends IKibanaSearchResponse>(
-    strategy: ISearchStrategy<Request, Response>,
-    ...args: Parameters<ISearchStrategy<Request, Response>['search']>
-  ) => Observable<Response>;
-  [prop: string]: any;
+export interface IScopedSearchSessionsClient<T = unknown> {
+  getId: (request: IKibanaSearchRequest, options: ISearchOptions) => Promise<string>;
+  trackId: (
+    request: IKibanaSearchRequest,
```

Comment:
It's funny  that it takes the request, even though it's scoped. 
Could we use it from the scoped client? :thinking:

Reply:
The request that `getId` and `trackId` take isn't the `KibanaRequest`, but rather the search request, so I think it still makes sense to include it here.

Reply:
:+1: